portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,21 +28,46 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  // adding hard-coded messages to arraylist
   ArrayList<String> sampleMessages = new ArrayList<>();
 
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
     String messages = convertToJsonString(sampleMessages);
     response.setContentType("text/html;");
     response.getWriter().println(messages);
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the inpiut from the form.
+    String userComment = getUserComment(request);
+    if(userComment == null) {
```

Comment:
I feel like you should throw an exception here. Not sure how the servlet works but shouldn't an exception return a non 200 status code to the client? You can bundle the error information along with that exception if thats the case.